arPullGlob.py

```diff
@@ -22,108 +24,71 @@
-            startTime = (time.mktime(datetime.datetime.startTime(strtTim, '%m/%d/%Y %H:%M:%S' ).timetuple()))
-            endTime = int(time.mktime(datetime.datetime.strptime(endTim, '%m/%d/%Y %H:%M:%S' ).timetuple()))
-            cavPv=makList()
-            codeFlt = ["0:FB_SUM", "0:RFS:INTLK_FIRST", "0:RFREADYFORBEAM"]
-            for pv in cavPv:
-                    cavPvFB.append(str(pv)+codeFlt[0])
-                    cavPvIntlk.append(str(pv)+codeFlt[1])
-                    cavPvRfRdy.append(str(pv)+codeFlt[2])
-            for pv in cavPv:
-                for i in range(3):
-                    pvList.append(str(pv) + codeFlt[i])
-            
-            url = self.url_formatter.format(SUFFIX=RANGE_RESULT_SUFFIX)
+    archiver=Archiver("lcls")
+
+    #global cavPvFB, cavPvIntlk, cavPvRfRdy
+    if (resultsFB == {}) or (strtTim != (time.mktime(datetime.datetime.strptime(strtTim, '%m/%d/%Y %H:%M:%S' ).timetuple()))) or (endTime != int(time.mktime(datetime.datetime.strptime(endTim, '%m/%d/%Y %H:%M:%S' ).timetuple()))):
+        print("We're in!")
+        startTime = datetime.datetime.strptime(strtTim, '%m/%d/%Y %H:%M:%S' )
+        endTime = datetime.datetime.strptime(endTim, '%m/%d/%Y %H:%M:%S' )
+        print('Start time {}'.format(startTime))
+        print('End time {}'.format(endTime))
+        cavPv=makList()
+# For after Sonya adds FB_SUM to the list
+#        codeFlt = ["FB_SUM", "RFS:INTLK_FIRST", "RFREADYFORBEAM"]
+        codeFlt = ["PHAFB_SUM", "RFS:INTLK_FIRST", "RFREADYFORBEAM"]
+        for pv in cavPv:
+            cavPvFB.append(str(pv)+codeFlt[0])
+            cavPvIntlk.append(str(pv)+codeFlt[1])
+            cavPvRfRdy.append(str(pv)+codeFlt[2])
+        for pv in cavPv:
+            for i in range(3):
+                pvList.append(str(pv) + codeFlt[i])
 
-            for pv in pvList:
-                       response = requests.get(url=url, timeout=TIMEOUT,
-                                               params={"pv": pv,
-                                                       "from": strtTim.isoformat() + "-07:00",
-                                                       "to": endTim.isoformat() + "-07:00"})
-                       results = {}
-    
-                       try:
-                           jsonData = json.loads(response.text)
-                           element = jsonData.pop()
-                           result = {"times": [], "values": []}
-                           for datum in element[u'data']:
-                               result["times"].append(datum[u'secs'])
-                               result["values"].append(datum[u'val'])
-    
-                           results[pv] = result["values"],result["times"]
-    
+        for pv in pvList:
+            print('PV: {}'.format(pv))
+            archiverData=archiver.getValuesOverTimeRange([pv],startTime,endTime)
```

Comment:
I think you can probably pull this call out of the for loop as `getValuesOverTimeRange(pvList, startTime, endTime)` unless I'm misreading it

Reply:
I'd like to run some speed tests. On matlab, surprisingly, calling get-archiver-data 3000 times is faster than calling it once with a list of 3000 pvs. I'd like to make sure the python code doesn't have the same "feature."